### PR TITLE
Fixes #10628.

### LINKED ISSUE #10628: bindSym + {.compileTime.} assign does not output a NimNode
When assigning a bindSym to a {.compileTime.} variable, bindSym does not output a NimNode.
It works in a static context.

Test case:

```Nim
import macros

type
  Input = object
    symbol: NimNode

proc input(symbol: NimNode): Input =
  Input(symbol: symbol)

# runtime value
import random
let x = rand(10)
echo "x: ", x

let y = rand(10)
echo "y: ", y

import typetraits
static: echo name(type(bindSym"x"))     # NimNode
# let x_sym{.compileTime.} = bindSym"x" # Cannot evaluate at compile time x
# static: echo x_sym.type.name

echo name(type(bindSym"x"))             # int

# Case 1 static block - works
static:
  let a = input(bindSym"x")
  let b = input(bindSym"y")

# Case 2 compileTime var - fails
block:
  let a{.compileTime.} = input(bindSym"x")
  let b{.compileTime.} = input(bindSym"y")

# Error: type mismatch: got <int>
# but expected one of:
# proc input(symbol: NimNode): Input

# expression: input(x)
```

### PR DESCRIPTION
**Pull request by deech**
Ensure a let or var decorated with the {.compileTime.} pragma runs in a static context.

### CODE REVIEW

File: compiler/semstmts.nim [hunk 1]
```diff
@@ -446,7 +446,16 @@ proc semVarOrLet(c: PContext, n: PNode, symkind: TSymKind): PNode =
       typ = semTypeNode(c, a.sons[length-2], nil)
 
     var def: PNode = c.graph.emptyNode
+    var compileTimeLet = false
     if a.sons[length-1].kind != nkEmpty:
+      if a.sons[0].kind == nkPragmaExpr:
```
Comment (Araq) [inline compiler/semstmts.nim]: Check for ``sfCompileTime in v.flags`` instead?

Comment (LemonBoy) [reply compiler/semstmts.nim]: `v` is only analyzed later. Doing a `pragma` scavenger hunt is definitely not the most elegant solution.

Comment (deech) [reply compiler/semstmts.nim]: If it works can it be merged with a TODO saying it needs to be refactored?

Comment (LemonBoy) [reply compiler/semstmts.nim]: Doing so is the textbook definition of how to accumulate [technical debt](https://en.wikipedia.org/wiki/Technical_debt), but that's just my opinion.

Comment (deech) [reply compiler/semstmts.nim]: That's a good point. I'll try and come up with a nicer way. I'm also open to suggestions as you know this code better than I do.

File: compiler/semstmts.nim [hunk 2]
```diff
@@ -449,13 +448,9 @@ proc semVarOrLet(c: PContext, n: PNode, symkind: TSymKind): PNode =
...
-        let pragmas = a.sons[0].sons[1]
-        for p in pragmas:
-          if p.kind == nkIdent and
-             whichKeyword(considerQuotedIdent(c,p)) == wCompileTime:
-            inc c.inStaticContext
-            compileTimeLet = true
-            break
+        var v = semIdentDef(c,a.sons[0],symKind)
+        if sfCompileTime in v.flags:
+          inc c.inStaticContext
```
Comment (deech) [inline compiler/semstmts.nim]: @LemonBoy Do you like this better?

Comment (deech) [reply compiler/semstmts.nim]: Actually that doesn't work with custom pragmas, reverting.

**Comment by deech** [comment]
I'm not sure why one CI is succeeding and the other isn't. Can someone run the appveyor one again?

**Comment by deech** [comment]
All checks pass now.

**Comment by deech** [comment]
Also not sure what commit etiquette is for this project so if you want me to squash them into one I'm happy to do so.

File: compiler/semstmts.nim (hunk 1 above)
Comment (Araq) [reply compiler/semstmts.nim]: Does that mean you didn't find a better way? :-/

Comment (deech) [reply compiler/semstmts.nim]: Yes. What I tried didn't work, I'm open to suggestions.

Comment (Araq) [reply compiler/semstmts.nim]: evaluate the value of the sym later after we evaluated its pragmas?

Comment (deech) [reply compiler/semstmts.nim]: Ok, I'll try.

**Comment by krux02** [comment]
@deech don't worry about squashing. Github has a button for that, you don't need to do anything for that.

**Pull request closed without merging**